File: server/router/system/sys_api_token.go
package system

import (
	"github.com/flipped-aurora/gin-vue-admin/server/api/v1"
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

type ApiTokenRouter struct{}

func (s *ApiTokenRouter) InitApiTokenRouter(Router *gin.RouterGroup) {
	apiTokenRouter := Router.Group("sysApiToken").Use(middleware.OperationRecord())
	apiTokenApi := v1.ApiGroupApp.SystemApiGroup.ApiTokenApi
	{
		apiTokenRouter.POST("createApiToken", apiTokenApi.CreateApiToken)   // 签发Token
		apiTokenRouter.POST("getApiTokenList", apiTokenApi.GetApiTokenList) // 获取列表
		apiTokenRouter.POST("deleteApiToken", apiTokenApi.DeleteApiToken)   // 作废Token
	}
}

File: web/src/api/sysApiToken.js
import service from '@/utils/request'

export const createApiToken = (data) => {
  return service({
    url: '/sysApiToken/createApiToken',
    method: 'post',
    data
  })
}

export const getApiTokenList = (data) => {
  return service({
    url: '/sysApiToken/getApiTokenList',
    method: 'post',
    data
  })
}

export const deleteApiToken = (data) => {
  return service({
    url: '/sysApiToken/deleteApiToken',
    method: 'post',
    data
  })
}
